js/upgrade.js

```diff
@@ -817,7 +817,7 @@ $(document).ready(function() {
         params.channel = "directory";
         params.directory_prestashop = $("select[name=directory_prestashop] option:selected").val();
         var directory_num = $("input[name=directory_num]").val();
-        if (directory_num == "" || directory_num.indexOf(".") == -1) {
+        if (directory_num == "" || !directory_num.match('^[0-9]{1}\\.[0-9]{1}\\.[0-9]{1}(\\.[0-9]{1})?$')) {
```

Comment:
Hi @FabienPapet and @MatShir 

The idea to force a valid version is a good one, however the test right doesn't seem strict enough. We actually need the whole version including the patch number because our upgrade scripts can vary depending on the patch target (for example we have a `1.7.8.1.sql` file to fix some bugs in the database).

But to apply the upgrade script correctly we really need the whole version which is not formatted the same way starting 8.0. So the regexp (or any other check) should take this into account and validate the input differently for vers 1.6.*, 1.7.* and 8.0.*

Reply:
Just an exmaple to illustrate more the problem With the current regexp the last part is optional, meaning if you input `1.7.5` it is considered valid and yet you don't have the patch version which will result in errors during the upgrade